### DISCUSSION TITLE
CAMARA Compliant: Expose all end points

### DISCUSSION THREAD

**Discussion by psidana1983** [camaraproject/DeviceStatus#72] [General]
Hi Team , 

As per the CAMARA documentation Device Status API has multiple end point URLs and If I implement only one end point then , can I still say that I am CAMARA compliant ?
 Or Is it mandatory to expose all the end points mentioned in CAMARA documentation to be the CAMARA compliant ?

one more question, assuming if I have implemented a current version this API and tomorrow if CAMARA decides to update this API with few more end points URLs then how it will get notified to us? Also is It  mandatory for us to implement the changes to be CAMARA compliant?

regards
Prashant Sidana

**Comment by psidana1983** [comment]
HI Team , 

Any updates?

Regards
Prashant Sidana

**Comment by hdamker** [comment]
Hi @psidana1983 Sorry that you haven't received an answer yet on your question. I assume that https://github.com/camaraproject/WorkingGroups/issues/305 is a duplicate of your question? If you have raised it in addition somewhere, let me know.

To your questions:

If there are multiple endpoints (operations) defined it would be good to implement them all. In some cases that might not possible (e.g. your network is not supporting a certain functionality). Then an appropriate return code and error message should be provided.  There could be also cases were you are not able to support certain parameters (e.g. you can't identify a device based on IPv4 addresses as you don't support them internally) - also here an appropriate return code and error message which helps the API consumer to understand the situation would be needed.

The second question, what about added endpoints in newer versions: your implementation is supporting a specific version of the API. That is still the case if there is already also a newer version of the API spec (e.g. we are currently supporting v0.8.1 of QoD API, and have v0.9.0 only in development). Supporting the next version requires to add the new endpoints/parameters to your provider implementation, so that an API client will get a proper behavior if sending requests according to the new API spec version. 

How do you learn about new versions: if you provide and expose an implementation of one or more CAMARA APIs you should follow at least the related mailing list. But we are working also on a coordinated release management and planning which will give you some informaiton ahead..

**Comment by psidana1983** [comment]
Hi Herbert ,
Thanks for your response . 

Correct it is a duplicate of what you have mentioned .
I had joined the Device Status issues discussion meeting where it was agreed to update the documentation with mandatory end point URL details so that implementer knows what is mandatory to be CAMARA compliant and I was asked to create a ticket in working groups which is why duplicate ticket is there. 



Regards
Prashant sidana

**Comment by psidana1983** [comment]
Hi Herbert , 

I have got one more question to ask 
As per the documentation , for device status API it is mandatory to return country code and country name in the response but as per my company policy I cannot share my clients location details so they are asking if I can remove country code/name from the response .

So I wanted to check If I can do this  and still be called CAMARA compliant .

Regards
Prashant Sidana
